subiquity/common/filesystem/tests/test_gaps.py

```diff
@@ -165,3 +167,153 @@ def test_half_extended_half_logical(self):
                 gaps.Gap(d, 35, 15, True),
                 gaps.Gap(d, 50, 50, False),
             ])
+
+
+class TestMovableTrailingPartitionsAndGapSize(unittest.TestCase):
+
+    def use_alignment_data(self, alignment_data):
+        m = mock.patch('subiquity.common.filesystem.gaps.parts_and_gaps')
+        p = m.start()
+        self.addCleanup(m.stop)
+        p.side_effect = partial(
+            gaps.find_disk_gaps_v2, info=alignment_data)
+
+    def test_no_next_gap(self):
+        self.use_alignment_data(PartitionAlignmentData(
+            part_align=10, min_gap_size=1, min_start_offset=10,
+            min_end_offset=10, primary_part_limit=10))
+        # 0----10---20---30---40---50---60---70---80---90---100
+        # #####[ p1                                   ]#####
+        m, d = make_model_and_disk(size=100)
+        p = make_partition(m, d, offset=10, size=80)
+        self.assertEqual(
+            ([], 0),
+            gaps.movable_trailing_partitions_and_gap_size(p))
+
+    def test_immediately_trailing_gap(self):
```

Comment:
These two tests in a row strike me as a bit odd.  In this test, p2 is not preserved, so because there is a gap before it, we can return the small gap as the answer.  In `test_one_trailing_movable_partition_and_gap`, we instead show the gap after p2, which only makes sense if we slide p2 to the end, but we could do that here also right?

I'm willing to accept 'follow-on branch' as an answer.

Reply:
Ah hah yes you are correct but

 1) I don't particularly want to think about the fact that in this case you might end up having to slide partitions by varying amounts
 2) fortunately I don't think there is or will be a way to create the test_immediately_trailing_gap situation, right? With this branch there's no way to create a hole without preserved partitions being around. So perhaps the correct response is to delete that test.